scripts/binary_to_array.py

```diff
@@ -0,0 +1,289 @@
+
+  data.extend([
+      "",
+      "extern const size_t %s;" % array_size_name,
+      "extern const unsigned char %s[];" % array_name,
+      "extern const char %s[];" % fileid,
+  ])
+
+  data.extend([
+      ""
+  ])
+  if namespaces:
+    data.extend([
+        "}  // namespace %s" % ns for ns in namespaces
+    ][::-1])  # close namespaces in reverse order
+  else:
+    data.extend([
+        "#if defined(__cplusplus)",
+        "}  // extern \"C\"",
+        "#endif  // defined(__cplusplus)"
+    ])
+  data.extend([
+      "",
+      "#endif  // %s" % header_guard,
+      ""
+  ])
+  return data
+
+
+def source(namespaces, array_name, array_size_name, fileid, filename,
+           input_bytes):
+  """Return a C/C++ source file for the given array.
+
+  Args:
+    namespaces: List of namespaces, outer to inner.
+    array_name: Name of the array.
+    array_size_name: Name of the array size constant.
+    fileid: Name of the identifier containing the filename.
+    filename: The original data filename itself.
+    input_bytes: Binary data to put into the array.
+
+  Returns:
+    A string containing the C/C++ source file.
+  """
+  data = []
+  data.extend([
+      "// Copyright 2016 Google Inc. All Rights Reserved.",
+      "",
+      "#include <stdlib.h>",
+      ""
+  ])
+  if namespaces:
+    data.extend([
+        "namespace %s {" % ns for ns in namespaces
+    ])
+  else:
+    data.extend([
+        "#if defined(__cplusplus)",
+        "extern \"C\" {",
+        "#endif  // defined(__cplusplus)"])
+
+  data.extend([
+      "",
+      "extern const size_t %s;" % array_size_name,
+      "extern const char %s[];" % fileid,
+      "extern const unsigned char %s[];" % array_name, "",
+      "const unsigned char %s[] = {" % array_name
+  ])
+  length = len(input_bytes)
+  line = ""
+  for idx in range(0, length):
+    if idx % WIDTH == 0:
+      line += "  "
+    else:
+      line += " "
+    line += "0x%02x," % input_bytes[idx]
+    if idx % WIDTH == WIDTH - 1:
+      data.append(line)
+      line = ""
+  data.append(line)
+  data.append("  0x00  // Extra \\0 to make it a C string")
+
+  data.extend([
+      "};",
+      "",
+      "const size_t %s =" % array_size_name,
+      "  sizeof(%s) - 1;" % array_name,
+      "",
+      "const char %s[] =" % fileid,
+      "  \"%s\";" % filename,
+      "",
+  ])
+
+  if namespaces:
+    data.extend([
+        "}  // namespace %s" % ns for ns in namespaces
+    ][::-1])  # close namespaces in reverse order
+  else:
+    data.extend([
+        "#if defined(__cplusplus)",
+        "}  // extern \"C\"",
+        "#endif  // defined(__cplusplus)"
+    ])
+  data.extend([
+      ""
+  ])
+  return data
+
+
+def main():
+  """Read an binary input file and output to a C/C++ source file as an array.
+
+  Raises:
+    Error: If an input file is not specified.
+  """
+
+  args = arg_parser.parse_args()
+
+  input_file = args.input
+  if not input_file:
+    raise Error("No input file specified.")
```

Comment:
Depending on how use parse the arguments, this check might go away. If it stays, then a builtin exception might be better. ValueError seems like it would fit.

Alternatively, just print an error and exit. It's not like anyone's going to catch this.

(If you're feeling particularly generous, in addition to printing the error, you could call `arg_parser.print_help()` or `.print_usage()` before exiting.)

Reply:
I think this should now be handled by `argparse`.